django/cantusdb_project/main_app/templates/chant_create.html

```diff
@@ -255,11 +255,59 @@ <h3>Create Chant</h3>
                 </div>
             </div>
+
+            {% if previous_chant %}
+                <div class="card w-100">
+                    <div class="card-body">
+                        <small>
+                            <b>Suggestions based on previous chant:</b><br>
+                            <div>
+                                {{ previous_chant.folio }} {{ previous_chant.c_sequence}} <a href="{% url 'chant-detail' previous_chant.id %}" target="_blank">{{ previous_chant.incipit }}</a><br>
+                                {% if previous_chant.cantus_id %}
+                                    (Cantus ID: <b><a href="https://cantusindex.org/id/{{ previous_chant.cantus_id }}" target="_blank">{{ previous_chant.cantus_id }}</a></b>)
+                                {% endif %}
+                            </div>
+                        {% if suggested_chants %}
+                            {% for suggestion in suggested_chants %}
+                                <input type="button" style="width: 80px" value="{{ suggestion.cantus_id }}" onclick="autoFill{{ suggestion.clean_cantus_id }}()" title="{{ suggestion.cantus_id }}"></input>
+                                <strong>{{ suggestion.genre_name }}</strong> - <span title="{{ suggestion.fulltext }}">{{ suggestion.incipit }}</span> (<strong>{{ suggestion.occurrences }}x</strong>)<br>
+                                <script>
+                                    function autoFill{{ suggestion.clean_cantus_id }}() {
+                                        document.getElementById('id_cantus_id').value = "{{ suggestion.cantus_id }}";
+                                        {% comment %} document.getElementById('id_genre').value = {{ suggestion.genre_id }}; {% endcomment %}
+                                        document.getElementById('id_manuscript_full_text_std_spelling').value = "{{ suggestion.fulltext }}";
+                                        var genre_id = {{ suggestion.genre_id | default_if_none:"null" }};
+                                        var genre_name = "{{ suggestion.genre_name }}";
+
+                                        // in case suggestion.genre_id is None, don't try to change the #id_genre selector
+                                        if (genre_id == null) {
+                                            return
+                                        }
+                                        // Set the value, creating a new option if necessary
+                                        if ($('#id_genre').find("option[value='" + genre_id + "']").length) {
+                                            $('#id_genre').val(genre_id).trigger('change');
+                                        } else { 
+                                            // Create a DOM Option and pre-select by default
+                                            var newOption = new Option(genre_name, genre_id, true, true);
```

Comment:
Why is this necessary with genres? What is the situation where we want a genre that doesn't exist as an option to be created?

Reply:
I've added comments to the places where we are setting both select elements.